docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,276 @@
+* CLI flags (e.g., `--FLAG`).
+* Environment variables following a consistent naming convention, such as `ILAB_<UPPERCASE_ARGUMENT_NAME>`.
+* Default values, for all the applicable use cases.
+
+For example, the `vectordb-uri` argument can be implemented using the `click` module like this:
+```py
+@click.option(
+    "--vectordb-uri",
+    default='rag-output.db',
+    envvar="ILAB_VECTORDB_URI",
+)
+```
+
+#### Local embedding models
+The embedding model used to generate text embeddings must be downloaded locally before executing the pipeline.
+
+For example, this command can be used to download the `sentence-transformers/all-minilm-l6-v2` model to the local models cache:
+```bash
+ilab model download -rp sentence-transformers/all-minilm-l6-v2
+```
+
+If the configured embedding model has not been cached, the command execution will terminate with an error. This requirement applies
+consistently to all new and updated commands.
+
+### 2.2 Document Processing Pipeline
+The proposal is to add a `process`  sub-command to the `data` command group:
+```
+ilab data process --input /path/to/docs/folder --output /path/to/processed/folder
+```
+
+#### Command Purpose
+Applies the transformation for the customer documents in `/path/to/docs/folder`. Processed artifacts are stored under `/path/to/processed/folder`.
+
+***Notes**: 
+* In alignment with the current SDG implementation, the folder will not be navigated recursively. Only files located at the root level of the specified
+folder will be considered. The same principle applies to all other options outlined below.
+* To ensure consistency and avoid issues with document versioning or outdated artifacts, the destination folder will be cleared before execution. 
+  This ensures it contains only the artifacts generated from the most recent run.
+
+The trasformation is based on the `instructlab-sdg` modules (the initial step of the `ilab data generate` pipeline)
+
+### Why We Need It
+This command streamlines the `ilab data generate` pipeline and eliminates the requirement to define a `qna` document,
+which typically includes:
+* A minimum of 5×3 question-and-answer pairs.
+* Reference documents stored in Git.
+
+The goal is not to generate training data for InstructLab-trained models but to utilize the documents for RAG 
+workflows with pre-tuned models.
+
+#### Usage
+The generated artifacts can later be used to generete and ingest the embeddings into a vector database.
+
+### 2.3 Document Processing Pipeline Options
+
+
+| Option Description | Default Value | CLI Flag | Environment Variable |
+|--------------------|---------------|----------|----------------------|
+| Base directories where models are stored. | `$HOME/.cache/instructlab/models`  | `--model-dir` | `ILAB_MODEL_DIR` |
+| Name of the embedding model. | **TBD** | `--embedding-model` | `ILAB_EMBEDDING_MODEL_NAME` |
+
+### 2.4 Embedding Ingestion Pipeline
+The proposal is to add an `ingest` sub-command to the `data` command group:
+```
```

Comment:
Must have the same default like the processing command: the `/path/to/docs/folder` argument is optional and, when missing, the processed artifacts are searched in the datasets folder.